cmd/server/app/server.go

```diff
@@ -42,17 +42,23 @@ import (
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/klog/v2"
-
 	"sigs.k8s.io/apiserver-network-proxy/cmd/server/app/options"
 	"sigs.k8s.io/apiserver-network-proxy/konnectivity-client/proto/client"
 	"sigs.k8s.io/apiserver-network-proxy/pkg/server"
+	"sigs.k8s.io/apiserver-network-proxy/pkg/server/leases"
 	"sigs.k8s.io/apiserver-network-proxy/pkg/util"
 	"sigs.k8s.io/apiserver-network-proxy/proto/agent"
 )
 
 var udsListenerLock sync.Mutex
 
-const ReadHeaderTimeout = 60 * time.Second
+const (
+	ReadHeaderTimeout    = 60 * time.Second
+	LeaseDuration        = 30 * time.Second
```

Comment:
GKE uses num agents ~= log(num nodes). So for 10K node clusters I don't expect a problem on GKE. However the ANP sample manifests uses a daemonset so num agents == num nodes. 10K agents renewing leases every 30 seconds would be an additional ~150 QPS of writes against the KAS. At some point we may want to A) use an alternate to daemonsets in our examples B) document the need for an alternate to dameonsets in our docs C) provide a configuration for the lease duration. Not saying it needs to be part of this PR but one of these should be a follow up AI.

Reply:
NM just realized this is server code not agent code. Deleting.